File: process_images.py
import numpy as np
from PIL import Image, ImageDraw
import glob
import os

files = glob.glob('src/assets/raw_icons/icon*.png')

# Target colors (RGB)
target_colors = [
    [255, 175, 204], # Pastel Pink
    [162, 210, 255], # Pastel Blue
    [253, 240, 213], # Soft Yellow
    [142, 202, 230], # Soft Turquoise
]

for idx, file in enumerate(files):
    img = Image.open(file).convert('RGBA')
    arr = np.array(img)
    
    r, g, b, a = arr[:,:,0], arr[:,:,1], arr[:,:,2], arr[:,:,3]
    
    # Mask for lime green. Green is usually higher than red and blue.
    mask = (g > 100) & (g > r + 15) & (g > b + 40)
    
    intensity = g[mask] / 200.0  # Normalize slightly higher to keep it bright
    
    target = target_colors[idx % len(target_colors)]
    
    arr[mask, 0] = np.clip(target[0] * intensity, 0, 255)
    arr[mask, 1] = np.clip(target[1] * intensity, 0, 255)
    arr[mask, 2] = np.clip(target[2] * intensity, 0, 255)
    
    out_img = Image.fromarray(arr)
    
    # Paint over Gemini watermark (bottom right corner)
    draw = ImageDraw.Draw(out_img)
    w, h = out_img.size
    
    # Cover the bottom right 10% of the corner, usually safe and hides the watermark
    box_size = int(w * 0.12)
    draw.rectangle([w-box_size, h-box_size, w, h], fill=(255, 255, 255, 255))
    
    out_name = file.replace('.png', '-pastel.png')
    out_img.save(out_name)
    print(f"Processed {file} -> {out_name}")

File: process_images.mjs
import { Jimp } from 'jimp';
import fs from 'fs';
import path from 'path';

const rawDir = 'src/assets/raw_icons';
const targetColors = [
    { r: 255, g: 175, b: 204 }, // Pink
    { r: 162, g: 210, b: 255 }, // Blue
    { r: 253, g: 240, b: 213 }, // Yellow
    { r: 142, g: 202, b: 230 }, // Turquoise
];

const files = fs.readdirSync(rawDir).filter(f => f.endsWith('.png') && !f.includes('-pastel'));

async function processImages() {
    for (let i = 0; i < files.length; i++) {
        const filePath = path.join(rawDir, files[i]);
        console.log(`Processing ${filePath}...`);
        try {
            const image = await Jimp.read(filePath);
            
            const target = targetColors[i % targetColors.length];
            const w = image.bitmap.width;
            const h = image.bitmap.height;
            const boxSize = Math.floor(w * 0.12);

            image.scan(0, 0, w, h, function(x, y, idx) {
                // Paint watermark corner white
                if (x >= w - boxSize && y >= h - boxSize) {
                    this.bitmap.data[idx + 0] = 255;
                    this.bitmap.data[idx + 1] = 255;
                    this.bitmap.data[idx + 2] = 255;
                    this.bitmap.data[idx + 3] = 255;
                    return;
                }

                const r = this.bitmap.data[idx + 0];
                const g = this.bitmap.data[idx + 1];
                const b = this.bitmap.data[idx + 2];
                
                // Mask for lime green: g > 100, g > r + 15, g > b + 40
                if (g > 100 && g > r + 15 && g > b + 40) {
                    const intensity = Math.min(g / 200.0, 1.2);
                    this.bitmap.data[idx + 0] = Math.min(255, Math.floor(target.r * intensity));
                    this.bitmap.data[idx + 1] = Math.min(255, Math.floor(target.g * intensity));
                    this.bitmap.data[idx + 2] = Math.min(255, Math.floor(target.b * intensity));
                }
            });
            
            const outName = filePath.replace('.png', '-pastel.png');
            await image.write(outName);
            console.log(`Saved ${outName}`);
        } catch (error) {
            console.error(`Error processing ${filePath}:`, error);
        }
    }
}

processImages();
